src/net/metaplaylist/isobmff_patcher.ts

```diff
@@ -0,0 +1,461 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    be4toi,
+    be8toi,
+    bytesToStr,
+    concat,
+    itobe4,
+    itobe8,
+    strToBytes,
+  } from "../../utils/bytes";
+
+/**
+ * Create a new _Atom_ (isobmff box).
+ * @param {string} name - The box name (e.g. sidx, moov, pssh etc.)
+ * @param {Uint8Array} buff - The box's content
+ */
+function Atom(name : string, buff : Uint8Array) : Uint8Array {
+  const len = buff.length + 8;
+  return concat(itobe4(len), strToBytes(name), buff);
+}
+
+export default class BoxPatchers {
+    private offset: number;
+    private lmsg: boolean;
+    private isTtml = false;
+    private segNr: number|undefined;
+    private topLevelBoxesToParse: string[] = [];
+    private compositeBoxesToParse: string[] = [];
+    private sizeChange = 0;
+    private tfdtValue: number = 0;
+    private duration: number = 0;
+    private ttmlSize: number = 0;
+    private data: Uint8Array;
+
+    constructor(
+        data: Uint8Array,
+        lmsg: boolean,
+        isTtml: boolean,
+        offset?: number,
+        segNr?: number
+    ) {
+        this.data = data;
+        this.topLevelBoxesToParse = ["moov", "styp", "sidx", "moof", "mdat"];
+        this.compositeBoxesToParse = ["trak", "moov", "moof", "traf"];
+        this.segNr = segNr;
+        this.offset = offset || 0;
+        this.lmsg = lmsg || false;
+        this.isTtml = isTtml || false;
+        if (this.isTtml) {
+            this.data = this.findAndProcessMdat(this.data);
+        }
+    }
+
+    /**
+     * Get size and boxtype from current box
+     * @param {Uint8Array} data
+     */
+    checkBox(data: Uint8Array) {
+      const size = be4toi(data, 0);
+      const boxtype = bytesToStr(data.subarray(4,8));
+      return {
+        size,
+        boxtype,
+      };
+    }
+
+    /**
+     * Triggers box processing
+     */
+    filter() {
+      let output = new Uint8Array(0);
+      let pos = 0;
+      while(pos < this.data.length) {
+        const { size, boxtype } = this.checkBox(this.data.subarray(pos, pos + 8));
+        const boxdata = this.data.subarray(pos, pos + size);
+        const concatData = (this.topLevelBoxesToParse
+          .indexOf(boxtype) >= 0) ?
+            this.filterBox(boxtype, boxdata, output.length) :
+            boxdata;
+        output = concat(output, concatData);
+        pos += size;
+      }
+      return output;
+    }
+
+    /**
+     * Process specific box and its children
+     * @param {string} boxtype
+     * @param {Uint8Array} data
+     * @param {number} filePos
+     * @param {string} _path
+     */
+    filterBox(
+      boxtype: string,
+      data: Uint8Array,
+      filePos: number,
+      _path?: string
+    ): Uint8Array{
+      let output = new Uint8Array(0);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this useless assignment to local variable "output". ([typescript:S1854](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS1854))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=178&issues=AWNyk0oJTmDVX2bfxyat&open=AWNyk0oJTmDVX2bfxyat)

[ISSUE_KEY]: AWNyk0oJTmDVX2bfxyat